web/src/message_helper.ts

```diff
@@ -25,22 +25,57 @@ export type NewMessage =
           raw_message: LocalMessage;
       };
 
-export function process_new_message(opts: NewMessage): Message {
+export type ProcessedLocalMessage = Message & {
+    queue_id: string | null;
+    to: string;
+    local_id: string;
+    topic: string;
+    draft_id: string;
+};
+
+export type ProcessedMessage =
+    | {
+          type: "server_message";
+          message: Message;
+      }
+    | {
+          type: "local_message";
+          message: ProcessedLocalMessage;
+      };
+
+export function process_new_server_message(raw_message: RawMessage): Message {
+    return process_new_message({
+        type: "server_message",
+        raw_message,
+    }).message;
+}
+
+export function process_new_local_message(raw_message: LocalMessage): ProcessedLocalMessage {
```

Comment:
Reading this, I would be tempted to go with a naming scheme of `RawLocalMessage/LocalMessage` paralleling `RawMessage/Message`.

Reply:
I like that, I'll make that change.